core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -188,6 +191,17 @@ public ZenDiscovery(Settings settings, ThreadPool threadPool, TransportService t
             DISCOVERY_REJOIN_ACTION_NAME, RejoinClusterRequest::new, ThreadPool.Names.SAME, new RejoinClusterRequestHandler());
     }
 
+    // protected to allow overriding in tests
+    protected ZenPing newZenPing(Settings settings, ThreadPool threadPool, TransportService transportService,
+                                 UnicastHostsProvider hostsProvider) {
+        return new UnicastZenPing(settings, threadPool, transportService, hostsProvider);
+    }
+
+    // TODO: this shoudln't be public, clean up the zen ping interface somehow to better allow clearing for testing?
```

Comment:
> disruption tests do not use mock discovery

They don't use it because of the MockZenPings - if we make that an optional feature of MockZenDiscovery (via a setting or whatever), the disruption tests could use MockZenDiscovery, turning mock zen pings off.